lib/commands/help.rb

```diff
@@ -14,7 +14,41 @@ def prepare(data)
     end
 
     def run
-      all_command_info
+      <<~HEREDOC
+        #{report_issues}
+        #{join_the_channel}
+
+        #{command_descriptions}
+      HEREDOC
+    end
+
+    private
+
+    def command_descriptions
+      commands.
+        map(&:description).
+        compact.
+        sort.
+        join("\n")
+    end
+
+    def commands
+      Dir.glob("lib/commands/**/*.rb").
+        map { |path| File.basename(path).gsub(".rb", "") }.
+        map { |file_name| file_name.split("_").map { |slice| slice.capitalize }.join }.
+        map { |class_name| "Commands::#{class_name}" }.
+        map { |full_class_name| Object.const_get(full_class_name) }
```

Comment:
```
Dir["#{File.dirname(__FILE__)}/commands/**/*.rb"].each { |file| require file }
Commands.constants.map { |command| Object.const_get("Commands::#{command}").description  }
```

Shouldn't something like that do the job? 

```

May be we could even filtering out the command that don't have a description?
```
Dir["#{File.dirname(__FILE__)}/commands/**/*.rb"].each { |file| require file }
Commands.constants.map do |command|
    c =  Object.const_get("Commands::#{command}")
    c.description if blabla.respond_to?('description')
end
```

Reply:
@fabientownsend I like your solution a lot better than mine. Nice!

Reply:
I don't like the `respond_to?` check tough, I think I'd rather have non-implemented 'description` methods for some commands, because it makes the intent very clear.